src/appleseed.studio/mainwindow/project/attributeeditor.cpp

```diff
@@ -83,9 +85,12 @@ void AttributeEditor::edit(
         m_entity_editor.get(), SIGNAL(signal_applied(foundation::Dictionary)),
         receiver, slot_apply);
 
-    QObject::connect(
-        m_parent->window(), SIGNAL(signal_crop_window_cleared()),
-        m_entity_editor.get(), SLOT(slot_clear_crop_window_widget()));
+    FrameItem* frame_item = dynamic_cast<FrameItem*>(receiver);
```

Comment:
We can start from the top of the stack then to find the proper place for item determination.
So, the `AttributeEditor` change begins at [ProjectExplorer::slot_item_selection_changed()](https://github.com/appleseedhq/appleseed/blob/master/src/appleseed.studio/mainwindow/project/projectexplorer.cpp#L230).
Then, [ItemBase::slot_edit(AttributeEditor* attribute_editor = 0)](https://github.com/appleseedhq/appleseed/blob/master/src/appleseed.studio/mainwindow/project/itembase.h#L90) is called, the [FrameItem::slot_edit(AttributeEditor* attribute_editor)](https://github.com/appleseedhq/appleseed/blob/master/src/appleseed.studio/mainwindow/project/frameitem.cpp#L77) for our case.
[AttributeEditor::edit(...)](https://github.com/appleseedhq/appleseed/blob/master/src/appleseed.studio/mainwindow/project/attributeeditor.cpp#L65) is the last point of journey, since it spawns `EntityEditor`
And here's ~~Johnny~~ dilemma.
We can pass signals all the way down through hierarchy, until we will reach the 'AttributeEditor'. 
Since 'AttributeEditor' is not a `QObject`, it can only connect signals and slots for derivatives of `QObject` available at the scope, but can't have its own.
We either need to make `AttributeEditor` a `QObject` derivative, either add a getter method for `EntityEditor` instance.
As for the signal/slot transmission, we can try to determine Item type at [ProjectExplorer::slot_item_selection_changed()](https://github.com/appleseedhq/appleseed/blob/master/src/appleseed.studio/mainwindow/project/projectexplorer.cpp#L230). 
Other way is to create a virtual slot at `ItemBase`, keep it always connected, and implement all we need at `FrameItem`. Con - all items will know about event.